pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/ValidationManager.java

```diff
@@ -180,6 +192,22 @@ public void runValidation() {
     LOGGER.info("Validation completed");
   }
 
+  // If we add a new broker, we want to rebuild the broker resource.
+  void rebuildBrokerResourceWhenBrokerAdded(AbstractTableConfig tableConfig, IdealState brokerIdealState) {
+    String brokerTenantName = tableConfig.getTenantConfig().getBroker();
+    String tableName = tableConfig.getTableName();
+
+    Set<String> brokerTenantInstances = _pinotHelixResourceManager.getAllInstancesForBrokerTenant(brokerTenantName);
+    Set<String> idealStateBrokerInstances = brokerIdealState.getInstanceSet(tableName);
+
+    if(!idealStateBrokerInstances.equals(brokerTenantInstances)) {
+      LOGGER.info("Rebuilding broker resource for table {}", tableName);
+      _pinotHelixResourceManager.rebuildBrokerResourceFromHelixTags(tableName);
+    } else {
+      LOGGER.info("Broker resource is not rebuilt for table {}", tableName);
```

Comment:
There is no value add in seeing this log. This will probably cause more confusion. I suggest removing that.